src/main/kotlin/com/demonwav/mcdev/platform/mcp/debug/UngrabMouseDebugSessionListener.kt

```diff
@@ -30,18 +35,68 @@ class UngrabMouseDebugSessionListener(private val process: DebugProcessImpl) : X
         val virtualMachine = debugProcess.virtualMachineProxy as? VirtualMachineProxyImpl ?: return
         val evaluationContext = EvaluationContextImpl(suspendContextImpl, frameProxy)
 
-        val mouseClass = virtualMachine.classesByName("org.lwjgl.input.Mouse")?.singleOrNull() as? ClassType ?: return
+        val mouseClass = virtualMachine.classesByName("org.lwjgl.input.Mouse")?.singleOrNull() as? ClassType
+        // LWJGL 3 does not have the Mouse class, Minecraft uses its own MouseHelper instead
+        if (mouseClass != null) {
+            ungrab2(mouseClass, virtualMachine, debugProcess, evaluationContext)
+        } else {
+            ungrab3(virtualMachine, debugProcess, evaluationContext)
+        }
+    }
+
+    private fun ungrab2(
+        mouseClass: ClassType,
+        virtualMachine: VirtualMachineProxyImpl,
+        debugProcess: DebugProcessImpl,
+        evaluationContext: EvaluationContextImpl
+    ) {
         val isGrabbed = mouseClass.methodsByName("isGrabbed", "()Z")?.singleOrNull() ?: return
         val setGrabbed = mouseClass.methodsByName("setGrabbed", "(Z)V")?.singleOrNull() ?: return
-        val grabValue = virtualMachine.mirrorOf(grab)
+        val grabValue = virtualMachine.mirrorOf(false)
 
         val currentState = debugProcess.invokeMethod(evaluationContext, mouseClass, isGrabbed, emptyList())
-        if ((currentState as? BooleanValue)?.value() != grab) {
+        if ((currentState as? BooleanValue)?.value() == true) {
             debugProcess.invokeMethod(evaluationContext, mouseClass, setGrabbed, listOf(grabValue))
         }
     }
 
+    private fun ungrab3(
+        virtualMachine: VirtualMachineProxyImpl,
+        debugProcess: DebugProcessImpl,
+        evaluationContext: EvaluationContextImpl
+    ) {
+        val srgMap = modulePointer.module?.let { module ->
+            MinecraftFacet.getInstance(module, McpModuleType)?.srgManager?.srgMapNow
+        }
+
+        fun mcp(srg: String, mcp: String): String {
+            return srgMap?.getMcpClass(srg) ?: mcp
+        }
+
+        val minecraftClass = virtualMachine.classesByName("net.minecraft.client.Minecraft")
+            ?.singleOrNull() as? ClassType ?: return
+        val minecraftGetter =
+            minecraftClass.methodsByName(mcp("func_71410_x", "getInstance"), "()Lnet/minecraft/client/Minecraft;")
+                ?.singleOrNull() ?: return
+        val minecraft = debugProcess.invokeMethod(
+            evaluationContext,
+            minecraftClass,
+            minecraftGetter,
+            emptyList()
+        ) as? ObjectReference ?: return
+
+        val mouseHelperField = minecraftClass.fieldByName(mcp("field_71417_B", "mouseHelper")) ?: return
+        val mouseHelper = minecraft.getValue(mouseHelperField) as? ObjectReference ?: return
+
+        val mouseHelperClass =
+            virtualMachine.classesByName("net.minecraft.client.MouseHelper")?.singleOrNull() as? ClassType ?: return
```

Comment:
Is this necessary? I think you can just do `mouseHelper.referenceType()` instead. Will help to get rid of another hardcoded string (though this one is stable so shouldn't be an issue).